### ISSUE TITLE
Build instructions missing

### ISSUE THREAD

**Issue by almereyda** [josephbisch/vmbuilder#1]
It is not completely obvious on how to build this, in so it becomes available to the system as a command line tool.

**Comment by almereyda** [comment]
Also http://stackoverflow.com/a/18452363/1959568 applies.

**Comment by josephbisch** [comment]
This is a fork of an upstream project with the intention being to add support for creating Debian VMs. So, with reference to the stackoverflow link, that would be something to bring to upstream and talk to them about fixing such issues.

As for installing vmbuilder, you should do so in the same way you usually do for Python code that ships with a setup.py file.

**Comment by almereyda** [comment]
Thanks for the quick reply. This is definitely to mention.

The build path is interesting for debian in that way, that `setuptools` provides a `bdist_rpm`, but no `bdist_deb`. `python setup.py install` obviously helped out here.

Getting the conversation back to upstream could probably also help with https://github.com/mistio/mist.io/issues/886 . If that happens, I'd be ready to contribute via PRs, but lets wait for tension over there.

**Comment by josephbisch** [comment]
It seems like `bdist_deb` would require `distutils` and `stdeb`.

To clarify, this fork adds support for creating Debian guest VMs. It looks like you are trying to use an Ubuntu Trusty image in the mist.io issue you linked. So there isn't any particularly special reason to be using this fork over the upstream version, except for the potential convenience of this repo being available using `git`. Even if you are using Debian as your host OS, you can use `ubuntu-vm-builder`, because the `ubuntu-vm-builder` program essentially just runs vmbuilder with the Ubuntu plugin that allows you to create Ubuntu guest VMs. You should be able to install any dependencies even if you are running Debian.

**Comment by almereyda** [comment]
Thanks for the clarification.
Indeed I am aiming at both, Debian guests as well as hosts.
Anyhow I didn't find any deb packages or apt repositories - or are you rather suggesting directly using the ones from Ubuntu? Sounds hacky, but might work, yes.

**Comment by josephbisch** [comment]
No, I wasn't suggesting using the Ubuntu packages, I was just making sure you realized that this fork isn't necessary if you just want to run vmbuilder on Debian to build Ubuntu VMs. The only deb packages of any version of vmbuilder that I am aware of are the ones in the Ubuntu Archive and yes, those ones are built for Ubuntu. If you have a need to make a deb package, then you can try modifying setup.py to use distutils and use that in conjunction with stdeb.
